components/portal/messages/services.js

```diff
@@ -300,12 +300,60 @@ define(['angular'], function(angular) {
           // Update storage
           $localStorage.hasUnseenAnnouncements = hasAnnouncements;
         };
+        var getMessagesByTitle = function(messages) {
+          var promises = [];
+          var filteredMessages = [];
+          var toBeDiscarded = [];
+          angular.forEach(messages, function(message) {
+            if (message.titleUrl) {
+              promises.push($http.get(message.titleUrl)
+                .then(function(result) {
+                  if (result.data) {
+                     var titleObject = result.data;
+                     if (titleObject.result && titleObject.result.length > 0) {
+                       var fromApi = titleObject.result[0];
+                       if (fromApi.full) {
+                         message.title = fromApi.full;
```

Comment:
Documentation added. Since we're talking about refactoring this whole service anyway, it doesn't seem to make sense to go through the contortions to make it configurable.